src/compile/facet.ts

```diff
@@ -21,6 +21,17 @@ import {assembleLayout, parseFacetLayout} from './layout';
 import {Model} from './model';
 import {initScale, parseScaleComponent} from './scale';
 
+/**
+ * Prefix for special data sources for driving column's axis group.
+ */
+
+export const COLUMN_AXES_DATA_PREFIX = 'column-';
```

Comment:
We seem to be a bit inconsistent about whether we use - or _ for scale or data source names. I guess we can use - for data sources and _ for scales given that we use scales in expressions.